samples/shopping-cart-addview/src/main/java/shoppingcart/application/UserEntity.java

```diff
@@ -0,0 +1,62 @@
+package shoppingcart.application;
+
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import akka.Done;
+import akka.javasdk.annotations.ComponentId;
+import akka.javasdk.eventsourcedentity.EventSourcedEntity;
+import akka.javasdk.eventsourcedentity.EventSourcedEntityContext;
+import shoppingcart.domain.UserEvent;
+import shoppingcart.domain.UserState;
+
+// tag::entity[]
+@ComponentId("user")
+public class UserEntity extends EventSourcedEntity<UserState, UserEvent> {
+  private final String entityId;
+
+  private static final Logger logger = LoggerFactory.getLogger(UserEntity.class);
+
+  public record CloseCartCommand(String oldCartId, String newCartId) {
+  }
+
+  public UserEntity(EventSourcedEntityContext context) {
+    this.entityId = context.entityId();
+  }
+
+  public ReadOnlyEffect<String> currentCartId() {
+    return effects().reply(currentState().currentCartId());
+  }
+
+  // We don't create a random cart ID in this method because we always
+  // want command handlers to be deterministic
+  public Effect<Done> closeCart(CloseCartCommand command) {
+    // Reject close commands for anything other than the current cart ID -
+    // idempotent
+    if (!command.oldCartId().equals(currentState().currentCartId())) {
+      return effects().reply(Done.getInstance());
+    }
+    return effects()
+        .persist(new UserEvent.UserCartClosed(entityId, command.newCartId()))
+        .thenReply(__ -> Done.getInstance());
+  }
+
+  @Override
+  public UserState emptyState() {
+    UUID uuid = UUID.randomUUID();
```

Comment:
emptyState will be called many times. It will only stop after we close the first cart. 

That means, that if you add two items, they will be added to two different cart ids.

Reply:
We can solve this in two ways: 

* we can have a monotonic id for the cart, (int or long). Starting from 1, for example.
* the first cart id equals the user id

Reply:
I think I like the monotonic better because it's just a counter. We can't make it simpler than that.